Automattic/vip-go-mu-plugins
# Added limit to index version count

## Description

Currently there is no limit to the number of index versions an indexable can have. This is a bad idea, so we're adding a limit of 2 per indexable.

Also needed to adjust a lot of the tests due to the added restriction.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

1. Execute `lando wp vip-search index-versions add post` 3 or 4 times.
2. Apply PR.
3. Execute `lando wp vip-search index-versions add post`. You should get a WP_Error telling you you can't add any more.
4. Verify with `lando wp vip-search index-versions list post` that no other versions were added.
5. Delete all the index versions.
6. Execute `lando wp vip-search index-versions add post` until you get the warning from 3.
7. Verify you only have two versions with `lando wp vip-search index-versions list post`
8. Activate users feature and repeat steps with the user indexable.
